src/brod_producer.erl

```diff
@@ -144,16 +144,17 @@ start_link(ClientPid, Topic, Partition, Config) ->
 produce(Pid, Key, Value) ->
   CallRef = #brod_call_ref{ caller = self()
                           , callee = Pid
-                          , ref    = make_ref()
+                          , ref    = Mref = erlang:monitor(process, Pid)
                           },
-  ok = gen_server:cast(Pid, {produce, CallRef, Key, Value}),
-  ExpectedReply = #brod_produce_reply{ call_ref = CallRef
-                                     , result   = brod_produce_req_buffered
-                                     },
-  %% Wait until the request is buffered
-  case sync_produce_request(ExpectedReply) of
-    ok              -> {ok, CallRef};
-    {error, Reason} -> {error, Reason}
+  Pid ! {produce, CallRef, Key, Value},
+  receive
```

Comment:
http://www.javalimit.com/2010/08/erlang-message-queues-and-rpc.html seems to be an old but good reference on the issue. It's unfortunate that it doesn't seem to be something that comes up in any of the general erlang performance things and only by knowing the name recv_mark was I able to find some information about it.